salt/tests/unit/modules/test_cordon.py

```diff
@@ -0,0 +1,101 @@
+import os.path
+import yaml
+
+from parameterized import parameterized
+
+from salt.exceptions import CommandExecutionError
+
+from salttesting.mixins import LoaderModuleMockMixin
+from salttesting.unit import TestCase
+from salttesting.mock import MagicMock, patch
+
+import metalk8s_cordon
+
+
+YAML_TESTS_FILE = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    "files", "test_metalk8s_cordon.yaml"
+)
+
+
+class CordonTestCase(TestCase, LoaderModuleMockMixin):
+    """
+    TestCase for `cordon` module
+    """
+    loader_module = metalk8s_cordon
+
+    def test_virtual(self):
+        """
+        Tests the return of `__virtual__` function
+        """
+        self.assertEqual(metalk8s_cordon.__virtual__(), 'metalk8s_kubernetes')
+
+
+    @parameterized.expand([
+        (
+          "bootstrap",
+          yaml.safe_load(open(YAML_TESTS_FILE))["cordon_object_result"]
+        ),
+    ])
```

Comment:
node is not really usefull here, you should have 2 tests for cordon, one with unschedulable already set to `True` and another one with not